src/component.js

```diff
@@ -125,6 +125,7 @@ export function enqueueRender(c) {
 /** Flush the render queue by rerendering all queued components */
 function process() {
 	let p;
+	q.sort((a, b) => a._depth - b._depth);
```

Comment:
Hmm we should consider splicing `q` (e.g. `q.splice(0)` or something similar) so that any `setState` calls queued up in `forceUpdate` don't get added to `q` after it has been sorted. I believe that any `setState` calls invoked while processing a previous `setState` should be rerendered after the current `setState` finished (i.e. should be added at the "end" of the queue, which for us means processed in another pass through the queue).

Reply:
Gonna see what the react behavior is here and perhaps that can become a test?

Reply:
I have added this change, good spot. Makes me wonder if we should propagate depth in diffChildren aswell....

Reply:
With splice it seems to fail a bunch of tests

Reply:
I had a go at this, but it seems to be a bit more involved. As far as I understand splicing the queue would demand a way to know the cutoff point at which the first item that was not part of the original queue is processed, so that we can sort the second update individually.

Another way could be to append components to the queue as as usual, but to resort it before the next call to `pop()` in case there were any new items added.

The third way I can come up with is to use a separate array all together. That is my least preferred option, because it can get really tricky when a component is in both queues and so on.

But then again I'm wondering if we don't already support that. Whenever we trigger an update and something calls `forceUpdate`, like during a lifecycle, we immediately update the component's children. Those are then flagged via `c._dirty` so that the queue will skip processing them. I'm probably missing something, but to me it seems to work out well (at least in my head).

Either way I feel like this requires more thought and would be better to work on in another PR. It will likely brew some discussions and that would leave the issue open for longer.